contrib/rtlbrowse/logfile.c

```diff
@@ -93,1390 +89,1308 @@ static GtkTextTag *blue_tag = NULL, *fwht_tag = NULL;
+        JRB node = jrb_find_int(lx2vals, *pnt_facidx);
+        Jval jv;
+
+        if (node) {
+            free(node->val.s);
+            node->val.s = strdup(*pnt_value);
+        } else {
+            jv.s = strdup(*pnt_value);
+            jrb_insert_int(lx2vals, *pnt_facidx, jv);
+        }
+    }
 }
 
-
-static gboolean
-button_release_event (GtkWidget *text, GdkEventButton *event)
+// When double-clicking an identifier in logfile.
+// Try to interpert it as a sub-module of the current module.
+// If such module exist. Open it in a new tab.
+static void import_doubleclick(GtkWidget *text, char *s)
 {
-(void)event;
+    struct text_find_t *t = text_root;
+    char *s2;
+    ds_Tree *ft = flattened_mod_list_root;
 
-gchar *sel;
+    while (t) {
+        if (text == t->text) {
+            break;
+        }
+        t = t->next;
+    }
 
-GtkTextIter start;
-GtkTextIter end;
+    if (!t)
+        return;
 
-if (gtk_text_buffer_get_selection_bounds (gtk_text_view_get_buffer(GTK_TEXT_VIEW(text)),
-                                         &start, &end))
-       {
-       if(gtk_text_iter_compare (&start, &end) < 0)
-               {
-               sel = gtk_text_buffer_get_text(gtk_text_view_get_buffer(GTK_TEXT_VIEW(text)),
-                                              &start, &end, FALSE);
+    s2 = malloc(strlen(t->ctx->which->fullname) + 1 + strlen(s) + 1);
+    sprintf(s2, "%s.%s", t->ctx->which->fullname, s);
 
-		if(sel)
-			{
-			if(strlen(sel))
-			{
-			int i, len = strlen(sel);
-			char *sel2;
-			char ch;
-
-			for(i=0;i<len;i++)
-				{
-				if(!is_identifier(sel[i])) goto bail;
-				}
-
-			while(gtk_text_iter_backward_char(&start))
-				{
-		               	sel2 = gtk_text_buffer_get_text(gtk_text_view_get_buffer(GTK_TEXT_VIEW(text)),
-                                              &start, &end, FALSE);
-				if(!sel2) break;
-				ch = *sel2;
-				g_free(sel2);
-				if(!is_identifier(ch))
-					{
-					gtk_text_iter_forward_char(&start);
-					break;
-					}
-				}
-
-			gtk_text_iter_backward_char(&end);
-			for(;;)
-				{
-				gtk_text_iter_forward_char(&end);
-		               	sel2 = gtk_text_buffer_get_text(gtk_text_view_get_buffer(GTK_TEXT_VIEW(text)),
-                                              	&start, &end, FALSE);
-				if(!sel2) break;
-				ch = *(sel2 + strlen(sel2) - 1);
-				g_free(sel2);
-				if(!is_identifier(ch))
-					{
-					gtk_text_iter_backward_char(&end);
-					break;
-					}
-				}
-
-		 	sel2 = gtk_text_buffer_get_text(gtk_text_view_get_buffer(GTK_TEXT_VIEW(text)),
-						&start, &end, FALSE);
-
-
-			/* oec->set_selection(oe, lft, rgh); */
-
-			import_doubleclick(text, sel2);
-			g_free(sel2);
-			}
-bail:			g_free(sel);
-			}
-		}
-	}
+    while (ft) {
+        if (!strcmp(s2, ft->fullname)) {
+            bwlogbox(ft->fullname, 640 + 8 * 8, ft, 0);
+            break;
+        }
 
+        ft = ft->next_flat;
+    }
 
-return(FALSE); /* call remaining handlers... */
+    free(s2);
```

Comment:
To match the change above:
```suggestion
    g_free(s2);
```